src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/MobDropListener.java

```diff
@@ -31,11 +33,18 @@ public void onEntityKill(EntityDeathEvent e) {
         if (e.getEntity().getKiller() != null) {
             Player p = e.getEntity().getKiller();
             ItemStack item = p.getInventory().getItemInMainHand();
-
+            Random rnd = new Random();
+            
             Set<ItemStack> customDrops = SlimefunPlugin.getRegistry().getMobDrops(e.getEntityType());
-            if (customDrops != null && !customDrops.isEmpty()) {
-                addDrops(p, customDrops, e.getDrops());
-            }
+            if (customDrops != null && !customDrops.isEmpty()) 
+            	for(ItemStack is:customDrops)
+            	{
+            		SlimefunItem sfi = SlimefunItem.getByItem(is);
+            		if(sfi != null && sfi instanceof ChanceDrop) 
```

Comment:
Also: An instanceof check always returns false for null, so the null check is redundant here.